drivers/power/supply/act8945a.c

```diff
@@ -0,0 +1,872 @@
+
+  config.frequency = priv->i2c_freq;
+  config.address   = priv->i2c_addr;
+  config.addrlen   = 7;
+
+  /* Write the register address */
+
+  flags = spin_lock_irqsave(NULL);
+  ret = i2c_write(priv->i2c, &config, &regaddr, sizeof(regaddr));
+  spin_unlock_irqrestore(NULL, flags);
+  if (ret < 0)
+    {
+      act8945a_err("ERROR: i2c_write failed: %d\n", ret);
+      return ret;
+    }
+
+  /* Restart and read 8 bits from the register */
+
+  flags = spin_lock_irqsave(NULL);
+  ret = i2c_read(priv->i2c, &config, regval, sizeof(*regval));
+  spin_unlock_irqrestore(NULL, flags);
+  if (ret < 0)
+    {
+      act8945a_err("ERROR: i2c_read failed: %d\n", ret);
+      return ret;
+    }
+
+  act8945a_info("INFO: addr: %02x value: %02x\n", regaddr, *regval);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: act8945a_putreg
+ *
+ * Description:
+ *   Writes a single byte to one of the ACT8945A registers.
+ *
+ ****************************************************************************/
+
+static int act8945a_putreg(FAR struct regulator_act8945a_priv *priv,
+                           uint8_t regaddr, uint8_t regval)
+{
+  struct i2c_config_s config;
+  uint8_t             buffer[2];
+  int                 ret;
+  irqstate_t          flags;
+
+  /* Sanity check */
+
+  DEBUGASSERT(priv != NULL);
+
+  /* Set up a 2-byte message to send */
+
+  buffer[0] = regaddr;
+  buffer[1] = regval;
+
+  /* Set up the I2C configuration */
+
+  config.frequency = priv->i2c_freq;
+  config.address   = priv->i2c_addr;
+  config.addrlen   = 7;
+
+  /* Write the register address followed by the data (no RESTART) */
+
+  flags = spin_lock_irqsave(NULL);
+  ret = i2c_write(priv->i2c, &config, buffer, sizeof(buffer));
+  spin_unlock_irqrestore(NULL, flags);
+
+  if (ret < 0)
+    {
+      act8945a_err("ERROR: i2c_write failed: %d\n", ret);
+      return ret;
+    }
+
+  act8945a_info("INFO:addr: %02x value: %02x\n", regaddr, regval);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: act8945a_list_voltage
+ *
+ * Description:
+ *
+ * Input Parameters:
+ *
+ *   rdev     - The regulator device pointer.
+ *   selector - The selector for the voltage look-up
+ *
+ * Returned Value: The voltage for the selector, or error code.
+ *
+ ****************************************************************************/
+
+static int act8945a_list_voltage(FAR struct regulator_dev_s *rdev,
+                                 unsigned int selector)
+{
+  if (selector < ACT8945A_NUM_VOLTAGES)
+    {
+      return (mv_list[selector]);
+    }
+  else
+    {
+      return -EINVAL;
+    }
+}
+
+/****************************************************************************
+ * Name: act8945a_enable_pulldown
+ *
+ * Description:
+ *
+ * Input Parameters:
+ *
+ *   rdev     - The regulator device pointer.
+ *
+ * Returned Value: Success or error code.
```

Comment:
Ditto: move Returned Value comment to new line instead putting everything in a single line.

Reply:
done